server/src/postgis/flight.rs

```diff
@@ -346,11 +350,207 @@ pub async fn get_flight_intersection_stmt(
+                ON flights."aircraft_identifier" = aircraft."identifier"
+            WHERE 
+                (
+                    -- get grounded aircraft without a scheduled flight
+                    ST_Intersects(ST_Envelope($1), aircraft."geom")
+                    AND aircraft."last_position_update" >= $2
+                    AND aircraft."last_position_update" <= $3
+                ) OR (
+                    -- get aircraft in flight
+                    flights."geom" IS NOT NULL
+                    AND ST_Intersects(ST_Envelope($1), flights."geom")
+                    AND flights."time_end" >= $2
+                    AND flights."time_start" <= $3
+                );
+            "#,
+            flights_table_name = get_flights_table_name(),
+            aircraft_table_name = super::aircraft::get_table_name(),
+        ))
+        .await
+        .map_err(|e| {
+            postgis_error!("(get_flights) could not prepare cached statement: {}", e);
+            FlightError::DBError
+        })?;
+
+    let result = client
+        .query(&stmt, &[&linestring, &time_start, &time_end])
+        .await
+        .map_err(|e| {
+            postgis_error!("(get_flights) could not execute transaction: {}", e);
+            FlightError::DBError
+        })?;
+
+    let mut flights = result
+        .iter()
+        .map(|row| {
+            let identifier: Option<String> = row.try_get(0)?;
+            let aircraft_id: String = row.try_get(1)?;
+            let aircraft_type: AircraftType = row.try_get(2)?;
+            let simulated: bool = row.try_get(3)?;
+
+            Ok(Flight {
+                identifier,
+                aircraft_id,
+                simulated,
+                positions: vec![],
+                state: None,
+                aircraft_type: aircraft_type as i32,
+            })
+        })
+        .collect::<Result<Vec<Flight>, tokio_postgres::error::Error>>()
+        .map_err(|e| {
+            postgis_error!("(get_flights) could not get flight data: {}", e);
+            FlightError::DBError
+        })?;
+
+    postgis_debug!("(get_flights) found {} flights.", flights.len());
+
+    // TODO(R5): Change this to use Redis 60s telemetry storage to acquire
+    //  telemetry information
+    let stmt = client
+        .prepare_cached(&format!(
+            r#"SELECT
+                    "geom",
+                    "velocity_horizontal_ground_mps",
+                    "velocity_vertical_mps",
+                    "track_angle_degrees",
+                    "last_position_update",
+                    "op_status"
+                FROM {table_name} 
+                WHERE "identifier" = $1
+                LIMIT 1;
+        "#,
+            table_name = super::aircraft::get_table_name(),
+        ))
+        .await
+        .map_err(|e| {
+            postgis_error!("(get_flights) could not prepare cached statement: {}", e);
+            FlightError::DBError
+        })?;
+
+    fn process_row(
```

Comment:
I don't think it's a good idea to rely on column order to get the row data.
You can get the data using the column name instead (see example in svc_storage: https://github.com/Arrow-air/svc-storage/blob/8baec86c3ae5233a36210810ccc5226cfa323a64/server/src/resources/flight_plan/mod.rs#L217)

Reply:
Great call, forgot this was possible

Reply:
https://github.com/Arrow-air/svc-gis/pull/19/commits/208ea9f19d9c58fd9d971792242d17f2a1b0b87e